src/collection-view.js

```diff
@@ -125,11 +125,75 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
-  // get the child view by model it holds, and remove it
-  _onCollectionRemove(model) {
-    const view = this.children.findByModel(model);
-    this.removeChildView(view);
-    this._checkEmpty();
+  // Handle collection update model removals
+  _onCollectionUpdate(collection, {changes: {removed = []} = {}} = {}) {
```

Comment:
actually, I am changing it to
```
// Handle collection update model removals
  _onCollectionUpdate(collection, options) {
    const changes = options.changes;
    if (changes) {
      const removed = changes.removed;
      if (_.isArray(removed)) {
        this._removeChildViews(removed);
      }
    }
  },
```
in case options doesn't have `options.changes` and only do stuff when removed is an array